esphome/components/slow_pwm/slow_pwm_output.cpp

```diff
@@ -21,20 +43,25 @@ void SlowPWMOutput::loop() {
   }
 
   if (scaled_state > now - this->period_start_time_) {
-    this->pin_->digital_write(true);
+    this->set_state_(true);
   } else {
-    this->pin_->digital_write(false);
+    this->set_state_(false);
   }
 }
 
 void SlowPWMOutput::dump_config() {
   ESP_LOGCONFIG(TAG, "Slow PWM Output:");
-  LOG_PIN("  Pin: ", this->pin_);
+  if (this->pin_)
+    LOG_PIN("  Using pin: ", this->pin_);
```

Comment:
The macro already has a condition for nullptr pins
```suggestion
  LOG_PIN("  Using pin: ", this->pin_);
```